lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -71,13 +71,19 @@ sub get_match_param {
 sub list_ajax {
     my ($self) = @_;
 
+    my $validation = $self->validation;
+    $validation->optional('limit')->num(0);
+    my $limit = $validation->param('limit');
+    return $self->render(json => {error => 'Limit exceeds maximum'}, status => 400)
+      unless $limit <= $self->app->config->{'global'}->{'jobs_results_limit'};
```

Comment:
these three lines are duplicated from the change in the API endpoint. Better extract into a function

Reply:
I don't know... do we want the same behavior everywhere? We don't have a `limit` everywhere

Reply:
I don't know if we want the same behaviour everywhere but to my eyes the above looks like a verbatim copy